Modules/audioop.c

```diff
@@ -1903,31 +1923,60 @@ static PyMethodDef audioop_methods[] = {
 };
 
 static int
-audioop_traverse(PyObject *m, visitproc visit, void *arg) {
-    _audioopstate *state = _audioopstate(m);
-    if (state != NULL)
+audioop_traverse(PyObject *m, visitproc visit, void *arg)
+{
+    audioop_state *state = (audioop_state *)PyModule_GetState(m);
+    if (state) {
         Py_VISIT(state->AudioopError);
+    }
     return 0;
 }
+
 static int
-audioop_clear(PyObject *m) {
-    _audioopstate *state = _audioopstate(m);
-    if (state != NULL)
+audioop_clear(PyObject *m)
+{
+    audioop_state *state = (audioop_state *)PyModule_GetState(m);
+    if (state) {
         Py_CLEAR(state->AudioopError);
+    }
     return 0;
 }
+
 static void
 audioop_free(void *m) {
     audioop_clear((PyObject *)m);
 }
 
+static int
+audioop_exec(PyObject* m)
+{
+    audioop_state *state = get_audioop_state(m);
+
+    state->AudioopError = PyErr_NewException("audioop.error", NULL, NULL);
+    if (state->AudioopError == NULL) {
+        return -1;
+    }
+
+    Py_INCREF(state->AudioopError);
+    if (PyModule_AddObject(m, "error", state->AudioopError) < 0) {
+        Py_DECREF(state->AudioopError);
+        return -1;
+    }
+
+    return 0;
+}
+
+static PyModuleDef_Slot audioop_slots[] = {
+    {Py_mod_exec, audioop_exec}
```

Comment:
I used a debugger on Windows. You forgot the terminator:

```suggestion
    {Py_mod_exec, audioop_exec},
    {0, NULL}
```

Reply:
Oh, good catch.
Thanks a million, victor.